ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -14,12 +14,25 @@ function join_by { local IFS="$1"; shift; echo "$*"; }
 
 REGION="${LEASED_RESOURCE}"
 
+workers=3
+if [[ "${SIZE_VARIANT}" == "compact" ]]; then
+  workers=0
+fi
+master_type=null
+if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+  master_type=m5.8xlarge
+elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+  master_type=m5.4xlarge
+elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+  master_type=m5.2xlarge
+fi
+
 # Generate working availability zones from the region
 mapfile -t AVAILABILITY_ZONES < <(aws --region "${REGION}" ec2 describe-availability-zones | jq -r '.AvailabilityZones[] | select(.State == "available") | .ZoneName' | sort -u)
-# Generate availability zones with the biggest instance type required
-mapfile -t XLARGE_ZONES < <(aws --region "${REGION}" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values=m5.8xlarge | jq -r '.InstanceTypeOfferings[].Location' | sort -u)
+# Generate availability zones with control and compute instance types required
+mapfile -t INSTANCE_ZONES < <(aws --region "${REGION}" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values="${master_type}","${COMPUTE_NODE_TYPE}" | jq -r '.InstanceTypeOfferings[].Location' | sort -u)
```

Comment:
We should include the bootstrap type, too, which is currently hardcorded in the installer to `m5.large`.